src/combinatorics/set_family/mathlib.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.upper_lower.basic
+import tactic.by_contra
+
+section linear_order
+variables {α : Type*} [linear_order α] {s t : set α}
+
+lemma is_lower_set.total (hs : is_lower_set s) (ht : is_lower_set t) : s ⊆ t ∨ t ⊆ s :=
+begin
+  by_contra' h,
+  simp_rw set.not_subset at h,
+  obtain ⟨⟨a, has, hat⟩, b, hbt, hbs⟩ := h,
+  obtain hab | hba := le_total a b,
+  { exact hat (ht hab hbt) },
+  { exact hbs (hs hba has) }
+end
+
+lemma is_upper_set.total (hs : is_upper_set s) (ht : is_upper_set t) : s ⊆ t ∨ t ⊆ s :=
+begin
+  by_contra' h,
+  simp_rw set.not_subset at h,
+  obtain ⟨⟨a, has, hat⟩, b, hbt, hbs⟩ := h,
+  obtain hab | hba := le_total a b,
+  { exact hbs (hs hab has) },
+  { exact hat (ht hba hbt) }
+end
+
+end linear_order
+
+namespace upper_set
+variables {α : Type*} [linear_order α]
+
+--TODO: Remove globally
+local attribute [-instance] set_like.partial_order
```

Comment:
I don't really agree. The intent of `set_like` was "has set-like membership, coe_sort, and inclusion" operators.
What you'd have to advocate for here is an additional weaker typeclass that doesn't provide it.

Reply:
What instance is it finding instead?

Reply:
`upper_set.partial_order`, which is the reverse order.